How-To/Create-a-New-Residential-Feature.md

```diff
@@ -0,0 +1,182 @@
+
+```         
+{{
+      config(          
+        materialized='table',         
+        partitioned_by=['year'],          
+        bucketed_by=['pin10'],          
+        bucket_count=1
+      )
+    }}
+```
+
+If the output does not have the partition column as the final column, you will receive the following error:
+
+```         
+HIVE_COLUMN_ORDER_MISMATCH: Partition keys must be the last columns in the table and in the same order as the table properties: 
+```
+
+-   Build out the transformation. To ensure consistency, use one of the macros whenever possible, each of which utilizes a specific coding structure. For example, the `dist_to_nearest_geometry.sql` requires a value for `pin`, `year`, `x_3435`, and `y_3435`. These also need to be structured in accordance with specific data types. In this case, if geographies are classified as a string, you will run into the following error.
+
+```         
+Runtime Error in model spatial.spatial_stadium (models/spatial/spatial.spatial_stadium.sql) line 9:5: Unknown type: STRING
+```
+
+To remedy this, you can recast the geometry value as a binary points.
+
+```         
+ST_ASBINARY(ST_POINT(stadium.lon, stadium.lat)) AS geometry
+```
+
+-   Once your model is defined correctly, run `dbt build` in your terminal to construct the output in your development database:
+
+```         
+dbt build --select proximity.dist_pin_to_stadium
+```
+
+-   Complete the documentation by configuring it as a [source](https://docs.getdbt.com/docs/build/sources) in the folder's dbt `schema.yml` file, and update the folder's `docs.md` and `columns.md` files.
+
+------------------------------------------------------------------------
+
+## Step 4: Incorporate the Transformed Data into the [Final Model View](https://github.com/ccao-data/data-architecture/tree/master/dbt/models/model).
+
+-   Now that the relevant view(s) have been created, they need to be incorporated in the model view.
+
+-   Identify if the new feature is relevant in the condo and/or the residential model. Then, in `dbt/models/model`, update the relevant `vw_pin_condo_input`, `vw_res_card_input`, or `vw_shared_input` view.
+
+-   If the attribute is spatial, you should also update `crosswalk_year_fill.sql`, `proximity.vw_pin10_proximity_fill.sql`, and `location.vw_pin10_location_fill.sql`. Make sure to fill in missing years of data: If certain years are missing from the data but we think that it did not change during that period, we can forward- or backward-fill it.
+
+------------------------------------------------------------------------
+
+## Step 5: Update the [Model Pipeline](https://github.com/ccao-data/model-res-avm/tree/master) to Use the New Versions of the Model Input Views
+
+-   Create a new branch in the `model-res-avm` repository.
+
+-   Add the new feature to the `predictor` section of `model/models/params.yaml`.
```

Comment:
**[Nitpick, blocking]** I think `params.yaml` lives at the top level of the model repo, right?

```suggestion
-   Add the new feature to the `predictor` section of `params.yaml`.
```